src/AutoRest.CSharp/Mgmt/Models/RequestPath.cs

```diff
@@ -202,6 +202,60 @@ public bool IsAncestorOf(RequestPath other)
         return true;
     }
 
+    /// <summary>
+    /// Check if <paramref name="requestPath"/> is a prefix path of <paramref name="candidate"/>
+    /// While comparing, we will ignore everything inside {}
+    /// For instance, if "/subs/{subsId}/rgs/{name}/foo" and "/subs/{subsId}/rgs/{name}/foo/bar/{something}",
+    /// we are effectively comparing /subs/{}/rgs/{}/foo and /subs/{}/rgs/{}/foo/bar/{}
+    /// </summary>
+    /// <param name="requestPath"></param>
+    /// <param name="candidate"></param>
+    /// <returns></returns>
+    public static bool IsPrefix(string requestPath, string candidate)
+    {
+        // Create spans for the candidate and request path
+        ReadOnlySpan<char> candidateSpan = candidate.AsSpan();
+        ReadOnlySpan<char> requestPathSpan = requestPath.AsSpan();
+
+        int cIdx = 0, rIdx = 0;
+
+        // iterate through everything on request path
+        while (rIdx < requestPathSpan.Length)
+        {
+            // if we run out of candidate, return false because request path here is effectively longer than candidate
+            if (cIdx >= candidateSpan.Length)
+                return false;
+
+            // if we hit a {
+            char c = candidateSpan[cIdx];
+            char r = requestPathSpan[rIdx];
+            switch (c == '{', r == '{')
```

Comment:
how about compacting the structure a bit like:
```csharp
if (c != r)
{
  return false;
}

if (c == '{')
{
  // doing the skipping until meet `}`
}
else
{
  cIdx++;
  rIdx++;
}
```

Reply:
I changed the switch to use `(c, r)` instead so that it is more readable. This kind of switch will be compiled into if statements anyway.